techmatters/webchat
# bug: fix friendlyName attribute in pre_engagement_data

## Description


- **Bug fix:** friendlyName attribute not being added to pre_engagement_data
For some reason, Twilio defaults the value of `friendlyName` to `Anonymous` on Webchat.  So when we try to get the customizable values from the bot's memory, as defined on PrepopulateKeys.json in Flex-plugin, `friendlyName` always returns `Anonymous`.

**Solution**
I defined a function that checks whether `friendlyName` is defined in the pre-engagement form. If it does, it overrides the `friendlyName` value on Context. With this, we can properly map the values of the pre-engagement form on Webchat to that defined in PrepopulateKeys.json.

### Checklist



### Related Issues
Fixes [CHI-1934](https://tech-matters.atlassian.net/browse/CHI-1934)

### Verification steps

- On webchat, define the data in the pre-engagement form (include `friendlyName`)
- On Flex-plugin, define the corresponding data in `ChildInformationTab.json` or `CallerInformationTab.json` as the case may be. Also, make sure it is properly mapped in PrepopulateKeys.json.
- Start a webchat chat
- Accept the incoming chat on Aselo.
- Verify that the right tab (Child or Caller) is correctly pre-populated.

<img width="1512" alt="Screenshot 2023-05-16 at 6 09 49 PM" src="https://github.com/techmatters/webchat/assets/40429848/ce607037-e7e0-45ca-b42e-7df67ccdf7a9">

<img width="646" alt="Screenshot 2023-05-16 at 6 12 08 PM" src="https://github.com/techmatters/webchat/assets/40429848/c2fa1328-03c3-4136-a582-25f6da0ef970">

For quick Verification, you can use the Linea Libra Helpline Staging. It has been configured to meet all of the steps outlined above. Just start a conversation, accept the incoming chat, and verify that the Child tab in correctly pre-populated.
